packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -3213,6 +3258,118 @@ function flushRenderPhaseStrictModeWarningsInDEV() {
   }
 }
 
+function commitDoubleInvokeEffectsInDEV(
+  fiber: Fiber,
+  hasPassiveEffects: boolean,
+) {
+  if (enableDoubleInvokingEffects) {
+    if (__DEV__) {
+      invokeLayoutEffectsUnmountInDEV(fiber);
+      if (hasPassiveEffects) {
+        invokePassiveEffectsUnmountInDEV(fiber);
+      }
+
+      invokeLayoutEffectsMountInDEV(fiber);
+      if (hasPassiveEffects) {
+        invokePassiveEffectsMountInDEV(fiber);
+      }
```

Comment:
I think we need to be setting the current debug Fiber here, in case any of the effects trigger dev warnings. This would mirror what we're doing in the work loop:
```suggestion
      setCurrentDebugFiberInDEV(fiber);
      invokeLayoutEffectsUnmountInDEV(fiber);
      if (hasPassiveEffects) {
        invokePassiveEffectsUnmountInDEV(fiber);
      }

      invokeLayoutEffectsMountInDEV(fiber);
      if (hasPassiveEffects) {
        invokePassiveEffectsMountInDEV(fiber);
      }
      resetCurrentDebugFiberInDEV();
```